### ISSUE TITLE
Invalid regex string containing scandinavian characters closes application abruptly

### ISSUE THREAD

**Issue by Geving** [notepad-plus-plus/notepad-plus-plus#10534]
### Description of the Issue
When doing a regex search with an invalid regex string containing the Scandinavian characters "æøå", the application closes immediately.
The currently open, but unsaved, document is lost upon opening NPP again.
(If the same invalid regex is used, but omitting the "æøå", the usual red error message about invalid regex is shown.)

### Steps to Reproduce the Issue
1. Open a new document
2. Add some text
3. Search for the following invalid regex expression:
`[a-zæøå'- ]*`

### Expected Behavior
An error stating that the regex is invalid.
(The way it behaves if you remove æøå from the faulty regex expression.)

### Actual Behavior
Application closes abruptly.

### Debug Information
Notepad++ v8.1.4   (32-bit)
Build time : Aug 21 2021 - 12:58:42
Path : C:\Program Files (x86)\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 2009
OS Build : 19043.1165
Current ANSI codepage : 1252
Plugins : ComparePlugin.dll CSScriptNpp.dll JSMinNPP.dll mimeTools.dll NppConverter.dll NppExport.dll NPPJSONViewer.dll PyNPP.dll XMLTools.dll

**Comment by dail8859** [comment]
Confirmed with my install as well.

```
Notepad++ v8.1.4   (64-bit)
Build time : Aug 21 2021 - 13:04:59
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : -noPlugin
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 1909
OS Build : 18363.1734
Current ANSI codepage : 1252
Plugins : none
```

**Comment by mere-human** [comment]
It crashes because of unhandled exception that is thrown while creating an error message.
[wstring_convert::from_bytes](https://en.cppreference.com/w/cpp/locale/wstring_convert/from_bytes) throws std::range_error in [s2ws](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/PowerEditor/src/MISC/Common/Common.cpp#L1222).
![image](https://user-images.githubusercontent.com/9664141/134489331-f0961f1b-1a5a-4135-bd43-c41ea50e74cb.png)


Definitely worth fixing.
Workarounds might be:
1. Remove the regex string from the error message. This is simple but makes it hard to understand the error.
2. Handle the error message wide string properly.

**Comment by Geving** [comment]
Seems to be fixed in v8.1.9.2, thanks!

> Notepad++ v8.1.9.2 bug-fixes:
> 
> 1.  Fix failure of erasing & saving server logs (of network drive) regression.
> 2.  Fix dark mode not working on Windows 10 2H22.
**> 3.  Fix crash in regex search.**
> 4.  Fix UDL folding breaking with delimiters issue if UNIX EOL applied.
> 5.  Fix UDL generic dark mode via UDL UI not working issue.
> 6.  Fix disabling alternative icons on tab in dark mode issue.
